auth/user_mgt_test.go

```diff
@@ -140,22 +140,118 @@ func TestGetUserByPhoneNumber(t *testing.T) {
 	}
 }
 
+func TestGetUserByProviderIDNotFound(t *testing.T) {
```

Comment:
Also test for empty providerID, empty providerUID etc

Reply:
See below; I added them to the TestInvalidGetUser() test. (Arguably, that could be 4 or 5 different tests, but they are at least very tightly related.)